src/Rings/mpoly-ideals.jl

```diff
@@ -2003,3 +2003,68 @@ function Base.show(io::IO, I::MPolyIdeal)
     print(io, "Ideal ", _get_generators_string_one_line(I))
   end
 end
+
+
+
+
+_pluecker_sgn(a::Vector{Int},b::Vector{Int},t::Int) = count(z->z>t,a)+count(z->z<t,b)
+    
+@doc raw"""
+    flag_pluecker_ideal(dimensions::Vector{Int},n::Int)
+
+Returns generators of the defining ideal for the flag variety $\text{Fl}((d_1,\dots,d_k),n)$, where $(d_1,\dots,d_k) =$`dimensions` denotes the rank, with $d_j\leq n-1$. That is, the space of $k$-step flags of linear subspaces $V_1\subset\dots\subset V_k$ in $\mathbb{C}^n$, where $\text{dim}(V_j) = d_{j}$.You can obtain the generators for the \emph{complete flag variety}$ by taking `dimensions` $=(1,\dots,n-1)$ and `n`$=n$. 
+
+# Example
+Complete flag variety $\text{Fl}((1,2,3),4)$. 
+```jldoctest
+julia> flag_pluecker_ideal([1,2,3],4)
+Ideal generated by
+  x[[1]]*x[[3, 4]] - x[[3]]*x[[1, 4]] + x[[4]]*x[[1, 3]]
+  -x[[1, 4]]*x[[2, 3]] + x[[2, 4]]*x[[1, 3]] - x[[3, 4]]*x[[1, 2]]
+  x[[2]]*x[[3, 4]] - x[[3]]*x[[2, 4]] + x[[4]]*x[[2, 3]]
+  x[[1]]*x[[2, 4]] - x[[2]]*x[[1, 4]] + x[[4]]*x[[1, 2]]
+  x[[1]]*x[[2, 3]] - x[[2]]*x[[1, 3]] + x[[3]]*x[[1, 2]]
+  x[[1]]*x[[2, 3, 4]] - x[[2]]*x[[1, 3, 4]] + x[[3]]*x[[1, 2, 4]] - x[[4]]*x[[1, 2, 3]]
+  -x[[1, 4]]*x[[1, 2, 3]] + x[[1, 3]]*x[[1, 2, 4]] - x[[1, 2]]*x[[1, 3, 4]]
+  -x[[2, 4]]*x[[1, 2, 3]] + x[[2, 3]]*x[[1, 2, 4]] - x[[1, 2]]*x[[2, 3, 4]]
+  -x[[3, 4]]*x[[1, 2, 3]] - x[[1, 3]]*x[[2, 3, 4]] + x[[2, 3]]*x[[1, 3, 4]]
+  -x[[1, 4]]*x[[2, 3, 4]] + x[[2, 4]]*x[[1, 3, 4]] - x[[3, 4]]*x[[1, 2, 4]]
+
+```
+# Example 2
+
+Flag variety $\text{Fl}((1,3),4)$.
+```jldoctest
+julia> flag_pluecker_ideal([1,3],4)
+Ideal generated by
+  x[[1]]*x[[2, 3, 4]] - x[[2]]*x[[1, 3, 4]] + x[[3]]*x[[1, 2, 4]] - x[[4]]*x[[1, 2, 3]]
+```
+"""
+function flag_pluecker_ideal(dimensions::Vector{Int},n::Int)
+    
+    N = collect(1:n)
+    
+    L = reduce(union,[subsets(N,d) for d in dimensions])
+    
+    L1 = reduce(union,[subsets(N,d-1) for d in dimensions])
+    
+    L2 = reduce(union,[subsets(N,d+1) for d in dimensions])
+    
+    R,x = polynomial_ring(QQ,"x"=>L)
+    
+    xdict = Dict{Vector{Int}, MPolyElem}([L[i] => x[i] for i in 1:length(L)])
+    
+    pairs = [[a,b] for a in L1, b in L2]
+    
+    X = [t for t in pairs if length(t[2])-length(t[1])>=2]
```

Comment:
Maybe combine `pairs` and `X` like this (untested, may need some tweaking but I hope the idea is clear)
```suggestion
    X = [(a,b) for a in L1, b in L2 if length(b)-length(a)>=2]
```